ui/src/widgets/ui-chart/UIChart.vue

```diff
@@ -383,8 +379,8 @@ export default {
 
                 this.chart.data.datasets.push(d)
             } else {
-                // have we seen this x-value before?
-                if (xIndex >= 0 && (this.props.xAxisType === 'category' || this.props.xAxisType === 'radial')) {
```

Comment:
Gayan, can you clarify why `this.props.xAxisType === 'category' || this.props.xAxisType === 'radial'` was removed from the `if` statement please?

Reply:
The condition `this.props.xAxisType === 'category' || this.props.xAxisType === 'radial'` was removed to ensure that the `xIndex >= 0` check applies universally to all axis types. This change makes sure that data points are updated correctly for any axis type, not just category or radial.

Reply:
@Steve-Mcl thank you for pointing out. I reverted this line of change as it's not necessary for this change